draft/0004-user-interaction.md

```diff
@@ -0,0 +1,497 @@
+
+## 3. Introduction
+
+A complete user interaction flow can be divided into following steps:
+
+1. **identification**: The user needs to provide a unique `identifier` to the platform. This can be a username, email address, or third-party social identity. It will be used to uniquely identify the user in the system.
+2. **verification**: The `verification` process is used to verify the user's identity by asking for additional security information. This can be a password, a one-time code, or a passkey. Upon the security settings multiple verification methods can be used.
+3. **profile completion**: The user may be prompted to complete their `profile` and `mfa` settings during a interaction session. This can be a first-time user registration, a mandatory profile and `mfa` settings update for exiting users, or a additional third-party identity linking.
+4. **authentication**: The last step is the `authentication` process. The user will be authenticated and granted access to the platform by submitting current interaction session once all the mandatory security checks are resolved.
+
+### 3.1 Terminology
+
+- **sign-in experience(SIE)**: A user's experience when interacting with Logto platform. Including the front-end UI and the back-end API calls.
+- **experience APIs**: A set of APIs that allow end-users to interact with the Logto for identity verification and profile management.
+- **interaction session**: A session that is initiated when a user starts interacting with the Logto platform. The session will be terminated once the user logs out or the session expires.
+- **experience app**: The front-end application that provides the user interface for the user interaction flow.
+- **social/Single Sign-On(SSO) identity**: A third-party identity provider that allows users to sign in to the Logto platform using their social media or enterprise SSO accounts.
+- **identifier**: A piece of information that can be used to identify a user in the Logto system. For example, username, email, phone number, or social/SSO identity.
+- **verification record**: A `verification record` is a group of security data can be used to verify a user's identity by such as password, verification code, or TOTP code.
+- **verification code**: A one-time code sent to the user's email or phone number for identity verification.
+- **verification id**: The unique identifier for a verification record. Generated by the Logto server when a verification flow is initiated. User need to provide the verification id along with additional security information to verify their identity. The verification id can also be used to track the verification record status within the interaction session.
+- **Multi-Factor Authentication(MFA)**: A security process that requires more than one verification factor to verify a user's identity.
```

Comment:
can you use Sentence case for terms since they locate in the beginning of bullet points. e.g.:

- **Sign-in experience**: Blah blah